src/datasets/arrow_dataset.py

```diff
@@ -707,6 +707,43 @@ def unique(self, column: str) -> List[Any]:
 
         return self._data.column(column).unique().to_pylist()
 
+    @fingerprint_transform(inplace=False)
```

Comment:
```suggestion
```
Since you're reusing methods that are fingerprint transforms, you don't need to make `class_encode_column` a fingerprint transform ;)
The fingerprint will be properly updated anyway